packages/cli/src/cmds/beacon/handler.ts

```diff
@@ -43,6 +45,27 @@ export async function beaconHandler(args: BeaconArgs & GlobalArgs): Promise<void
   const abortController = new AbortController();
 
   logger.info("Lodestar", {network, version, commit});
+
+  const defaultThreadpoolSize = os.availableParallelism();
+  logger.info(`BLS libuv pool size: ${process.env.UV_THREADPOOL_SIZE}`);
+  if (isNaN(parseInt(`${process.env.UV_THREADPOOL_SIZE}`))) {
+    logger.warn(`UV_THREADPOOL_SIZE must be a number. Using default value of ${defaultThreadpoolSize}`);
+    process.env.UV_THREADPOOL_SIZE = defaultThreadpoolSize.toString();
+  }
+  /**
+   * Help users ensure that thread pool is large enough for optimal performance
+   *
+   * Node reports available CPUs. There is enough idle time on the main and
+   * network threads that setting UV_THREADPOOL_SIZE to $(nproc) provides the
+   * best performance. Recommend this value to consumers
+   */
+  if (parseInt(`${process.env.UV_THREADPOOL_SIZE}`) < defaultThreadpoolSize) {
```

Comment:
should `parseInt` once, assign to a variable to reuse

Reply:
add it to `logger.warn` below too

Reply:
Updated